setup.py

```diff
@@ -333,6 +335,7 @@ def do_setup():
             'all': devel_all,
             'devel_ci': devel_ci,
             'all_dbs': all_dbs,
+            'api': api_packages,
```

Comment:
```suggestion
            'rest': api_packages,
```
or
```suggestion
            'rest_api': api_packages,
```
What do you think?

Reply:
Sure, I've no objections

Reply:
Is this actually optional? Will the webserver start without this package installed?

Reply:
@ashb to answer your question, is the existing rest api depended on by the webserver?

Reply:
It's mounted as routes in the webserver, and always available I thought? I'm not sure anymore!

Reply:
I think you're right, yes, it's loaded as routes in the webserver.(https://github.com/apache/airflow/blob/82c60a2040a3f306e5f880340c3f646d4f629884/airflow/www/api/experimental/endpoints.py#L44)  I'm proposing moving all the API logic into a single module, and exposing it using the existing Airflow plugin architecture (https://github.com/apache/airflow/pull/4640/files#diff-5ff8468ade348aeb2ccc273cf3b79550R32)

This would be a require extra *only if* the existing UI or CLI requires the API to exist. If that's the case, we'd just add the questioned dependencies to the core, and remove the `[api]` extra flag from the entry_points.